src/utils/term.ts

```diff
@@ -0,0 +1,14 @@
+export class Term {
+
+    public term : string;
```

Comment:
I would create a new Term object for each synonym.The ID will be same for all of them (Group 1 assigns no IDs to synonyms).